include/mbedtls/oid.h

```diff
@@ -466,6 +471,25 @@ typedef struct mbedtls_oid_descriptor_t {
  */
 int mbedtls_oid_get_numeric_string(char *buf, size_t size, const mbedtls_asn1_buf *oid);
 
+/**
+ * \brief           Translate a string containing a dotted-decimal
+ *                  representation of an ASN.1 OID into its encoded form
+ *                  (e.g. "1.2.840.113549" into "\x2A\x86\x48\x86\xF7\x0D").
+ *                  On success, this function allocates oid->buf from the
+ *                  heap. It must be freed by the caller using mbedtls_free().
+ *
+ * \param oid       mbedtls_asn1_buf to populate with the DER-encoded OID
+ * \param oid_str   string representation of the OID to parse
+ * \param size      length of the OID string
```

Comment:
If we're passing the size (length?) of the string, does it need to be null-terminated? Does the length of the string include the terminator (i.e. is the size of the string) or not (i.e. the length of the string)

Reply:
Insisting on a null terminator can be safer in some scenarios, but in other scenarios it can be very annoying: if the OID is a substring of a larger string, or if it's a string allocated in a high-level language that doesn't do null termination. So I prefer not to require a null terminator. But if other similar functions in the library already want a null terminator, it's better to be consistent.

Reply:
I've opted for non-null-terminated strings because it is likely we will need to use this function to parse an OID from a larger string, and it's easier to do that in-place rather than copying it out.